library/src/containers/AsyncApi/Layout.tsx

```diff
@@ -25,12 +25,18 @@ const AsyncApiLayout: React.FunctionComponent<Props> = ({
   config,
   error = null,
 }) => {
-  const [divWidth, setDivWidth] = useState(1281);
+  const [observerClassName, setObserverClassName] = useState('container:xl');
 
   const { ref } = useResizeObserver<HTMLDivElement>({
     onResize: ({ width }) => {
-      if (divWidth !== width) {
-        width && setDivWidth(width);
+      if (width === undefined) {
+        return;
+      }
+
+      const possibleClassName =
+        width <= 1280 ? 'container:xl' : 'container:base';
+      if (possibleClassName !== observerClassName) {
+        setObserverClassName(possibleClassName);
```

Comment:
Setting the class instead of the width works already much better for me, but I still receive the error message (see [here](https://github.com/asyncapi/asyncapi-react/issues/498#issuecomment-998794808) ).

I only got around it by using a `setTimeout()`, thus it is not in the same cycle anymore.

```suggestion
        setTimeout(() => setObserverClassName(possibleClassName), 0);
```

Reply:
I wrapped callback with `requestAnimationFrame`, it fixed problem in my local env, so I think that is also a good option, but thanks for the suggestion! :)